main.go

```diff
@@ -177,50 +182,68 @@ func (m *Message) Bytes() ([]byte, error) {
 		header[k] = v
 	}
 
-	// Top level multipart writer for our `multipart/mixed` body.
-	mixedw := multipart.NewWriter(buffer)
-
 	header.Add("MIME-Version", "1.0")
-	header.Add("Content-Type", fmt.Sprintf("multipart/mixed;%s boundary=%s", crlf, mixedw.Boundary()))
-
-	err = writeHeader(buffer, header)
-	if err != nil {
-		return nil, err
-	}
 
-	// Write the start of our `multipart/mixed` body.
-	_, err = fmt.Fprintf(buffer, "--%s%s", mixedw.Boundary(), crlf)
-	if err != nil {
-		return nil, err
+	var mixedw *multipart.Writer
+	var hasAttachments = m.Attachments != nil && len(m.Attachments) > 0
+	if hasAttachments {
+		// Top level multipart writer for our `multipart/mixed` body.
+		// only needed if we have attachments
+		mixedw = multipart.NewWriter(buffer)
+		header.Add("Content-Type", fmt.Sprintf("multipart/mixed;%s boundary=%s", crlf, mixedw.Boundary()))
+		err = writeHeader(buffer, header)
+		if err != nil {
+			return nil, err
+		}
+		header = textproto.MIMEHeader{}
+		// Write the start of our `multipart/mixed` body.
+		_, err = fmt.Fprintf(buffer, "--%s%s", mixedw.Boundary(), crlf)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	// Does the message have a body?
 	if m.Body != "" || m.HTMLBody != "" {
```

Comment:
You're right, but actually I think it would be better to allow both to be empty, and just create a message with an empty plain text body and no html body.
This would also be backwards compatible.